internal/resolution/pm/composer/cmd_factory.go

```diff
@@ -0,0 +1,45 @@
+package composer
+
+import (
+	"os/exec"
+	"path/filepath"
+)
+
+type ICmdFactory interface {
+	MakeInstallCmd(command string, file string) (*exec.Cmd, error)
+}
+
+type IExecPath interface {
+	LookPath(file string) (string, error)
+}
+
+type ExecPath struct {
+}
+
+func (ExecPath) LookPath(file string) (string, error) {
+	return exec.LookPath(file)
+}
+
+type CmdFactory struct {
+	execPath IExecPath
+}
+
+func (cmdf CmdFactory) MakeInstallCmd(command string, file string) (*exec.Cmd, error) {
+	path, err := cmdf.execPath.LookPath(command)
+
+	fileDir := filepath.Dir(file)
+
+	return &exec.Cmd{
+		Path: path,
+		Args: []string{command, "update",
+			"--no-interaction",       // We can't answer any prompts...
+			"--no-scripts",           // Avoid risky scripts
+			"--ignore-platform-reqs", // We won't run the code, so we don't care about the platform
+			"--no-autoloader",        // We won't execute any code, no need for autoloader
+			"--no-install",           // No need to install packages
+			"--no-plugins",           // We won't run the code, so no plugins needed
+			"--no-audit",             // We don't want to run an audit
+		},
+		Dir: fileDir,
```

Comment:
So now we run the command in the directory of the file, and not on the specific file, right? 
Can there be multiple composer manifest files in the same directory? 

For instance, in Nuget it is common to have multiple .csproj and python multiple different requiremetns-*.txt etc.

Reply:
so.. should we run the command on the exact file instead?

Reply:
Apparently, specifying a specific composer.json file is using COMPOSER env variable, see https://getcomposer.org/doc/03-cli.md#composer

Reply:
I have never seen it, but since it is supported, I think it would be good for us to add it. Don't think it will be hard to add a env variable when executing the command?

Reply:
I think it can result in some flaky behavior, as multiple files are resolved at the same time.. creating a race condition. 
I vote skip until we get the feedback